### ISSUE TITLE
Possibility to read properties file passed in command line

### ISSUE THREAD

**Issue by eximius313** [novoda/gradle-build-properties-plugin#27]
It is very common situation that you have ```application.properties``` file, but on specific environment you must override it (take ```--spring.config.location=file://override.properties``` as an example).

It would be great if gradle-build-properties-plugin would support overriding properties from the command line too

**Comment by jarekCC** [comment]
I Agree with eximus313. It would be fantastic to have possibility to override properties from the command line

**Comment by mr-archano** [comment]
Hi @eximius313, it's not fully documented because not released yet, but with the generic support to maps now the plugin can support override at command line like shown in [this PR](https://github.com/novoda/gradle-build-properties-plugin/pull/26):
```
buildProperties {
    cli {
        from project.properties  
    }
}
```
which can be used later in your buildscript as
```
buildProperties.cli['overridable'].or(buildProperties.secrets['overridable'])
```

the early version of the plugin [supported the override from command line](https://github.com/novoda/gradle-build-properties-plugin/pull/12) like you mentioned but we decided we want to be **explicit** when it comes to resolution of properties. If not happy with that you can easily script a custom implementation of `Entries` that looks for properties at command line 
before evaluating the properties from a file or else. I can try to draft something later and see if meets your needs.

**Comment by eximius313** [comment]
I'm not that fluent in Grdle yet, so I think I don't fully understand your example.
Could you please show me on this particular case? I have:
```
buildProperties {
    app {
        file project(':backend').file('src/main/resources/application.properties')
    }
}
```
And I'm running springBoot as:
```
./gradlew -Dspring.config.location=/full/path/override.properties bootRun
```
And because ```override.properties``` contains database access, I must override it in both Spring and Gradle.
In SpringBoot it just works. How can I override my ```buildProperties``` in Gradle using your example?

Thank you in advance

**Comment by mr-archano** [comment]
Sorry for the confusion. What I meant is you can create another entry in the `buildProperties` closure:
```
buildProperties {
    app {
        file project(':backend').file('src/main/resources/application.properties')
    }
    cli {
        from project.properties
    }
}
```

and then you can define the fallback strategy as you need, for instance:
```
buildProperties.cli['overridable'].or(buildProperties.secrets['overridable'])
```
where it will check whether you have provided a `overridable` property at command line (eg: `./gradlew -Poverridable=foo ...`)
Which ultimately is **not** what you're asking I guess. My bad.

If I understand your scenario correctly you can just do:
```
buildProperties {
    app {
        def overridePath = System.properties['spring.config.location']
        if (overridePath) {
             file project.file(overridePath)
        } else {
             file project(':backend').file('src/main/resources/application.properties')
        }
   }
}
```
which you should be able to use with 
```
./gradlew -Dspring.config.location=/full/path/override.properties bootRun
```

**Comment by eximius313** [comment]
Almost!
As far as I can see, the perfect solution would look like this (pseudocode):
```
buildProperties {
    app {
        def overridePath = System.properties['spring.config.location']
        file project(':backend').file('src/main/resources/application.properties').override(project.file(overridePath))
   }
}
```
where ```override``` would take all properties in ```application.properties``` and override theese with the one found in ```project.file(overridePath)``` (override.properties can be a subset of application properties)

**Comment by mr-archano** [comment]
OH! Now I get it. you could use the built-in inheritance mechanism for properties files then (see [`README`](https://github.com/novoda/gradle-build-properties-plugin/blob/master/README.md#properties-inheritance)). In the file you use to override you define a `include=/path/to/application.properties` and then all the values you want to override.

I am also planning of removing this mechanism because not too flexible (relative paths provided by `include` property are now evaluated using the properties file [parent folder as base](https://github.com/novoda/gradle-build-properties-plugin/blob/master/plugin/src/main/groovy/com/novoda/buildproperties/FilePropertiesEntries.groovy#L18)). I really want to put in place something more explicit, similar to what you posted above, but embracing the inheritance of build properties holistically, eg:
```
buildProperties {
   default {
        file project(':backend').file('src/main/resources/application.properties')
    }
    app {
       include 'default'
       def overridePath = System.properties['spring.config.location']
       if (overridePath) {
             file project.file(overridePath) // properties with matching keys are overridden
        }
    }
```

But I'm not sure about the DSL yet. I'll consider the available options.

**Comment by mr-archano** [comment]
@eximius313 slowly, but this new feature is coming:
- #44 
- #45

**Comment by eximius313** [comment]
Great!! I'm eagerly waiting!
